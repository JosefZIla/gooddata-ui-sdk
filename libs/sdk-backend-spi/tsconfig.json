{
    "compilerOptions": {
        "esModuleInterop": true,
        "target": "es2017",
        "module": "commonjs",
        "lib": ["es2017", "webworker"],
        "moduleResolution": "node",
        "importHelpers": true,
        "sourceMap": true,
        "outDir": "./dist",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "removeComments": false,
        "baseUrl": "..",
        "paths": {
            "@gooddata/sdk-model": ["sdk-model/src"]
        }
    },
    "exclude": ["**/node_modules", "dist", "esm"]
}
