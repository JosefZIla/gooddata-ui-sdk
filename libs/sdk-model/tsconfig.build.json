{
    "extends": "./tsconfig.json",
    "include": ["src/index.ts", "src/dashboard/drillUrl.ts"],
    "compilerOptions": {
        "baseUrl": null,
        "paths": null,
        "declaration": true,
        "declarationMap": true
    }
}
