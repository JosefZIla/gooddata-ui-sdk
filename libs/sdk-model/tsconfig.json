{
    "compilerOptions": {
        "esModuleInterop": true,
        "target": "es5",
        "module": "commonjs",
        "moduleResolution": "node",
        "importHelpers": true,
        "lib": ["es6", "dom", "ES2016.Array.Include"],
        "declaration": true,
        "declarationMap": true,
        "sourceMap": true,
        "outDir": "./dist",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "removeComments": false
    },
    "exclude": ["**/node_modules", "dist", "esm"]
}
