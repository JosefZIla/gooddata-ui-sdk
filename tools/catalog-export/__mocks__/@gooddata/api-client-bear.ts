// (C) 2007-2018 GoodData Corporation
import catalogItemsResponse from "./loadItems.json";
import getObjectsResponse from "./getObjects.json";
import getObjectsByQueryResponse from "./getObjectsByQuery.json";
import getVisualizationsResponse from "./getVisualizations.json";
import getAnalyticalDashboards from "./getAnalyticalDashboards.json";

export default {
    catalogue: {
        loadItems() {
            return Promise.resolve(catalogItemsResponse);
        },
    },
    md: {
        getObjectsByQuery() {
            return Promise.resolve(getObjectsByQueryResponse);
        },
        getObjects() {
            return Promise.resolve(getObjectsResponse);
        },
        getVisualizations() {
            return Promise.resolve(getVisualizationsResponse);
        },
        getAnalyticalDashboards() {
            return Promise.resolve(getAnalyticalDashboards);
        },
    },
};
