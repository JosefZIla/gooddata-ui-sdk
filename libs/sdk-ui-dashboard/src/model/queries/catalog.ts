// (C) 2022 GoodData Corporation
import { IAttribute, IMeasure, ObjRef } from "@gooddata/sdk-model";
import { IDashboardQuery } from "./base";

/**
 * Query to obtain attribute catalog object according to given attribute reference.
 *
 * @public
 */
export interface QueryCatalogAttributes extends IDashboardQuery {
    readonly type: "GDC.DASH/QUERY.CATALOG.ATTRIBUTES";
    readonly payload: {
        readonly attributesOrRefs: ObjRef[] | IAttribute[];
    };
}

export function queryCatalogAttributes(
    attributesOrRefs: ObjRef[] | IAttribute[],
    correlationId?: string,
): QueryCatalogAttributes {
    return {
        type: "GDC.DASH/QUERY.CATALOG.ATTRIBUTES",
        correlationId,
        payload: {
            attributesOrRefs,
        },
    };
}

export interface QueryCatalogFacts extends IDashboardQuery {
    readonly type: "GDC.DASH/QUERY.CATALOG.FACTS";
    readonly payload: {
        factRefs: ObjRef[];
    };
}

export function queryCatalogFacts(factRefs: ObjRef[], correlationId?: string): QueryCatalogFacts {
    return {
        type: "GDC.DASH/QUERY.CATALOG.FACTS",
        correlationId,
        payload: {
            factRefs,
        },
    };
}

export interface QueryCatalogMeasures extends IDashboardQuery {
    readonly type: "GDC.DASH/QUERY.CATALOG.MEASURES";
    readonly payload: {
        measuresOrRefs: ObjRef[] | IMeasure[];
    };
}

export function queryCatalogMeasures(
    measuresOrRefs: ObjRef[] | IMeasure[],
    correlationId?: string,
): QueryCatalogMeasures {
    return {
        type: "GDC.DASH/QUERY.CATALOG.MEASURES",
        correlationId,
        payload: {
            measuresOrRefs,
        },
    };
}

export interface QueryCatalogDateDatasets extends IDashboardQuery {
    readonly type: "GDC.DASH/QUERY.CATALOG.DATE.DATASETS";
    readonly payload: {
        refs: ObjRef[];
    };
}

export function queryCatalogDateDatasets(refs: ObjRef[], correlationId?: string): QueryCatalogDateDatasets {
    return {
        type: "GDC.DASH/QUERY.CATALOG.DATE.DATASETS",
        correlationId,
        payload: {
            refs,
        },
    };
}
