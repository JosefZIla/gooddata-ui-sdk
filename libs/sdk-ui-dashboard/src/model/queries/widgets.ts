// (C) 2021 GoodData Corporation

import { IFilter, ObjRef } from "@gooddata/sdk-model";
import { IDashboardQuery } from "./base";

/**
 * Given a reference to an insight widget, this query will obtain the filters that should be used when executing the insight.
 * These will respect the ignored filters on widget level as well as the filters specified in the insight itself.
 * Filters returned by this query should be used with {@link @gooddata/sdk-model#insightSetFilters} to obtain
 * insight that is ready for execution.
 *
 * @alpha
 */
export interface QueryInsightWidgetFilters extends IDashboardQuery<IFilter[]> {
    readonly type: "GDC.DASH/QUERY.INSIGHT_WIDGET.FILTERS";
    readonly payload: {
        readonly widgetRef: ObjRef;
    };
}

/**
 * Creates action thought which you can query dashboard component for filters that should be used by a given insight widget.
 *
 * @param widgetRef - reference to insight widget
 * @param correlationId - optionally specify correlation id to use for this command. this will be included in all
 *  events that will be emitted during the command processing
 * @alpha
 */
export function queryInsightWidgetFilters(
    widgetRef: ObjRef,
    correlationId?: string,
): QueryInsightWidgetFilters {
    return {
        type: "GDC.DASH/QUERY.INSIGHT_WIDGET.FILTERS",
        correlationId,
        payload: {
            widgetRef,
        },
    };
}
