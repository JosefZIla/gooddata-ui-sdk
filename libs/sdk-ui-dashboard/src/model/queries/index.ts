// (C) 2021 GoodData Corporation

import { QueryInsightAttributesMeta, QueryInsightDateDatasets } from "./insights";
import { QueryWidgetFilters } from "./widgets";

export { IDashboardQuery, DashboardQueryType } from "./base";
export {
    QueryInsightDateDatasets,
    InsightDateDatasets,
    queryDateDatasetsForInsight,
    QueryInsightAttributesMeta,
    InsightAttributesMeta,
    queryInsightAttributesMeta,
    insightSelectDateDataset,
} from "./insights";
export { QueryMeasureDateDatasets, queryDateDatasetsForMeasure, MeasureDateDatasets } from "./kpis";
export { QueryWidgetFilters as QueryInsightWidgetFilters, queryWidgetFilters } from "./widgets";

/**
 * @alpha
 */
export type DashboardQueries = QueryInsightDateDatasets | QueryInsightAttributesMeta | QueryWidgetFilters;
