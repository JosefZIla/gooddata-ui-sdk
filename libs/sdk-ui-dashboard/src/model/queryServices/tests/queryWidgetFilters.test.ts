// (C) 2021 GoodData Corporation
import { DashboardTester, preloadedTesterFactory } from "../../tests/DashboardTester";
import {
    FilterTestingDashboardIdentifier,
    FilterTestingDashboardWidgets,
} from "../../tests/Dashboard.fixtures";
import { queryWidgetFilters } from "../../queries";
import { IFilter } from "@gooddata/sdk-model";
import { IWidget } from "@gooddata/sdk-backend-spi";

describe("query widget filters", () => {
    let Tester: DashboardTester;
    beforeEach(preloadedTesterFactory((tester) => (Tester = tester), FilterTestingDashboardIdentifier));

    it.each<[string, IWidget]>([
        ["insight with no ignored filters", FilterTestingDashboardWidgets.NoIgnoredFilters.Insight],
        ["kpi with no ignored filters", FilterTestingDashboardWidgets.NoIgnoredFilters.Kpi],
        [
            "insight with ignored attribute filter",
            FilterTestingDashboardWidgets.IgnoredAttributeFilter.Insight,
        ],
        ["kpi with ignored attribute filter", FilterTestingDashboardWidgets.IgnoredAttributeFilter.Kpi],
        ["insight with ignored date filter", FilterTestingDashboardWidgets.IgnoredDateFilter.Insight],
        ["kpi with ignored date filter", FilterTestingDashboardWidgets.IgnoredDateFilter.Kpi],
    ])("should return filters for %s", async (_, widget) => {
        const result: IFilter[] = await Tester.query(queryWidgetFilters(widget));
        expect(result).toMatchSnapshot();
    });
});
