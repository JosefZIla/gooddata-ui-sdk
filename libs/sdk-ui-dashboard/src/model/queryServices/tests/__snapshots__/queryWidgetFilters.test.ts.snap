// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query widget filters should return filters for insight with ignored attribute filter 1`] = `
Array [
  Object {
    "absoluteDateFilter": Object {
      "dataSet": Object {
        "uri": "/gdc/md/referenceworkspace/obj/557",
      },
      "from": "2012-01-01",
      "to": "2019-01-01",
    },
  },
]
`;

exports[`query widget filters should return filters for insight with ignored date filter 1`] = `
Array [
  Object {
    "negativeAttributeFilter": Object {
      "displayForm": Object {
        "uri": "/gdc/md/referenceworkspace/obj/1089",
      },
      "notIn": Object {
        "uris": Array [
          "/gdc/md/referenceworkspace/obj/1088/elements?id=460490",
        ],
      },
    },
  },
]
`;

exports[`query widget filters should return filters for insight with no ignored filters 1`] = `
Array [
  Object {
    "absoluteDateFilter": Object {
      "dataSet": Object {
        "uri": "/gdc/md/referenceworkspace/obj/557",
      },
      "from": "2012-01-01",
      "to": "2019-01-01",
    },
  },
  Object {
    "negativeAttributeFilter": Object {
      "displayForm": Object {
        "uri": "/gdc/md/referenceworkspace/obj/1089",
      },
      "notIn": Object {
        "uris": Array [
          "/gdc/md/referenceworkspace/obj/1088/elements?id=460490",
        ],
      },
    },
  },
]
`;

exports[`query widget filters should return filters for kpi with ignored attribute filter 1`] = `
Array [
  Object {
    "absoluteDateFilter": Object {
      "dataSet": Object {
        "uri": "/gdc/md/referenceworkspace/obj/557",
      },
      "from": "2012-01-01",
      "to": "2019-01-01",
    },
  },
]
`;

exports[`query widget filters should return filters for kpi with ignored date filter 1`] = `
Array [
  Object {
    "negativeAttributeFilter": Object {
      "displayForm": Object {
        "uri": "/gdc/md/referenceworkspace/obj/1089",
      },
      "notIn": Object {
        "uris": Array [
          "/gdc/md/referenceworkspace/obj/1088/elements?id=460490",
        ],
      },
    },
  },
]
`;

exports[`query widget filters should return filters for kpi with no ignored filters 1`] = `
Array [
  Object {
    "absoluteDateFilter": Object {
      "dataSet": Object {
        "uri": "/gdc/md/referenceworkspace/obj/557",
      },
      "from": "2012-01-01",
      "to": "2019-01-01",
    },
  },
  Object {
    "negativeAttributeFilter": Object {
      "displayForm": Object {
        "uri": "/gdc/md/referenceworkspace/obj/1089",
      },
      "notIn": Object {
        "uris": Array [
          "/gdc/md/referenceworkspace/obj/1088/elements?id=460490",
        ],
      },
    },
  },
]
`;
